servant/servant.cabal

```diff
@@ -60,6 +60,7 @@ library
     , http-types            >= 0.8  && < 0.10
     , mtl                   >= 2.0  && < 2.3
     , mmorph                >= 1    && < 1.1
+    , semigroups            >= 0.16 && < 0.19
```

Comment:
avoiding `semigroups` is essentially impossible. We could have it optional for GHC 8.0+ though.
[`attoparsec`](http://hackage.haskell.org/package/attoparsec) depends on `semigroups`, for example, so it's in the transitive closure already.